packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m

```diff
@@ -88,7 +88,17 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNoti
   NSDictionary *remoteNotification = response.notification.request.content.userInfo;
   if (remoteNotification[@"gcm.message_id"]) {
     NSDictionary *notificationDict = [RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];
-    [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_notification_opened" body:notificationDict];
+
+    // create a merged dictionary that will hold the serialized userInfo data, as well as any actionIdentifier
+    NSMutableDictionary * mergedDict = [NSMutableDictionary dictionary];
```

Comment:
I wonder - and this could be me just not understanding Objective-C - do we need to dealloc this dictionary anywhere / somehow? Is it already handled or is it automagically cleaned up?
I have a feeling that it will be orphaned and result in a memory leak

Reply:
Funny I wondered the same thing, not sure what happens to the other `NSDictionary` instances and if there is any mem difference between `NSDictionary` and `NSMutableDictionary`.

There are alternatives to cloning this dict but I decided they were worse for code quality:
1) Modify `[RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];` to take another argument for `actionIdentifier` (they are siblings in the parent object). Decided against because I would be butchering a very specifically named function to do something else, and possibly break the larger fn that parses the notification.
2) Make `[RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];` return an NSMutableDict so I can add the `actionIdentifier` in without cloning. Almost did this but again, really hesitant to modify that other function as it is quite an important one for other handlers.

I think I could dealloc the `mergedDict` after firing `sendEventWithName`?

Also think `_initialNotification` needs to have `mergedDict` instead of `notificationDict` for cold starts to work

Reply:
> I think I could dealloc the mergedDict after firing sendEventWithName?

This idea scares me as I think the JS side (being async) could consume it at some indeterminate future time so we can't safely dealloc until we *know* the JS side consumed.

For that reason I'm leaning towards merging even without resolving this, and lodging a separate issue to investigate (effectively peeling this discussion off and punting on it for the moment)

> Also think _initialNotification needs to have mergedDict instead of notificationDict for cold starts to work

You've definitely got the use case here and a test app to exercise, I'll defer to your thinking here

Reply:
Gentle comment: this seems like an unresolved thought here